### ISSUE TITLE
empty dataset throws exception on fold-by

### ISSUE THREAD

**Issue by pdmct** [scicloj/tablecloth#8]
Tablecloth 4.04.1

Exception is thrown by fold-by if dataset has no rows.

It should just return the original empty dataset.

```
tablecloth> DS
_unnamed [3 2]:

| :a | :b |
|----|----|
|  1 | :a |
|  2 | :b |
|  9 | :c |
tablecloth> (api/select-rows DS (comp #(< % 1) :a))
_unnamed [0 2]:

| :a | :b |
|----|----|
tablecloth> (api/fold-by (api/select-rows DS (comp #(< % 1) :a)) :a)
Execution error (ArityException) at tablecloth.api.group-by/ungroup (group_by.clj:203).
Wrong number of args (0) passed to: tech.ml.dataset.base/concat

```

**Comment by genmeblog** [comment]
confirmed, fixing

**Comment by genmeblog** [comment]
fixed in 4.04.2 and 5.0-SNAPSHOT
